DSCResources/MSFT_xDhcpServerAuthorization/MSFT_xDhcpServerAuthorization.psm1

```diff
@@ -0,0 +1,144 @@
+Import-Module $PSScriptRoot\..\Helper.psm1 -Verbose:$false
+
+# Localized messages
+data LocalizedData
+{
+    # culture="en-US"
+    ConvertFrom-StringData @'
+ResolvingIPv4Address = Resolving local IPv4 IP addresses
+ResolvingHostname    = Resolving local hostname
+AuthorizingServer    = Authorizing DHCP Server '{0}' with IP address '{1}'
+UnauthorizingServer  = Unauthorizing DHCP Server '{0}' with IP address '{1}'
+ServerIsAuthorized  = DHCP Server '{0}' with IP address '{1}' IS authorized
+ServerNotAuthorized  = DHCP Server '{0}' with IP address '{1}' is NOT authorized
+'@
+}
+
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [Parameter(Mandatory)]
+        [ValidateSet('Present','Absent')]
+        [System.String] $Ensure,
+
+        [ValidateNotNullOrEmpty()]
+        [System.String] $DnsName = ( Get-Hostname ),
+
+        [ValidateNotNullOrEmpty()]
+        [System.String] $IPAddress = ( Get-IPv4Address | Select-Object -First 1 )
+    )
+    $targetResource = @{
+        DnsName = $DnsName;
+        IPAddress = $IPAddress;
+        Ensure = ''
+    }
+    if ($Ensure -eq 'Present') {
+        if (Test-TargetResource -Ensure $Ensure -DnsName $DnsName -IPAddress $IPAddress) {
+            $targetResource['Ensure'] = 'Present';
+        }
+        else { $targetResource['Ensure'] = 'Absent'; }
+    }
+    elseif ($Ensure -eq 'Absent') {
+        if (Test-TargetResource -Ensure $Ensure -DnsName $DnsName -IPAddress $IPAddress) {
+            $targetResource['Ensure'] = 'Absent';
```

Comment:
Suggestion:  I would suggest the pattern of just doing the get, in get-targetresource, and calling get from test to perform the tests.    You are doing the opposite.  doing the get in test, and calling test from get and inferring the information.  It should still work with this simple of resource, but it's more complex and difficult to read in my opinion.